### DISCUSSION TITLE
Performance of VIT-B/32 is worse than RN50 on CC3M

### DISCUSSION THREAD

**Discussion by JACKHAHA363** [mlfoundations/open_clip#56] [Q&A]
Here are my curves. RN50 roughly matchs the one shown in the repo, but the VITB/32 is worse. I am using the hyperparams from README. I am wondering could you also share the performance curves of ViTB/32 on CC?
<img width="370" alt="Screen Shot 2021-09-08 at 12 39 26 PM" src="https://user-images.githubusercontent.com/14231180/132549768-5fc7c2b0-a9f6-464b-85e3-5b5786e6908f.png">

**Comment by carlini** [comment]
ViT-B performed worse for us on CC than a RN50. I suspect (but can not prove) this is because there's not enough data and vision transformers appear more data hungry than resnets. I don't have the accuracy off hand, but this looks comparable to what we were seeing.

**Comment by rwightman** [comment]
This is expected and your numbers appear reasonable. In training quite a few models at the lower end recently, the ViT-B models (even the smaller ones) will underperform similar sized ResNet models for smaller data. This includes up to at least the 12-15M sample range as I was unable to push ViT-B-32 past RN50 on cc12m or yfcc15m. I feel the crossover point is probably in the 40-100M sample range but have not verified that.

One could possibly work around this by using a pretrained backbone for the vision tower. There is partial support for this right now in some (preliminary) support for `timm` models... 

One could modify a model config such as this to enable the `timm_model_pretrained` flag https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/model_configs/timm-vit_base_patch32_224.json

You'd then be starting with a vision tower pretrained on imagenet. It significantly speeds up reaching decent zero shot and eval rseults BUT I'd caution against using an imagenet pretrained backbone and doing zero_shot eval on Imagenet, you'd probably want an alternate zero shot test dataset

**Reply by kyleliang919** [reply]
@rwightman  Do you mind sharing some of the numbers for ViT-B-32 on yfcc15m? Somehow I never got it over 15% with 32 epochs and 32K batch size and default params from the paper. I am just wondering if there is something wrong with my set up.

**Comment by rwightman** [comment]
Moving to discussion for future reference

**Comment by kyleliang919** [comment]
@rwightman Not sure if it's still relevant. For ViT-B/32, if you train it with longer epochs, there might be a chance to push it beyond convnets on smaller data domain such as YFCC 15M. I am seeing close to 30 top-1 imagenet1k acc by training 512 epochs instead of 32.  However, because of the learning rate scheduler, it plateaued in the end.
